src/java/org/apache/cassandra/concurrent/SyncFutureTask.java

```diff
@@ -71,7 +71,11 @@ public void run()
         catch (Throwable t)
         {
             tryFailure(t);
-            ExecutionFailure.handle(t);
+            // A lot of exceptions are expected and will be handled by Cassandra
+            // by consuming the result of the future task so only treat Error
+            // as uncaught
+            if (t instanceof Error)
```

Comment:
These exceptions aren't uncaught they are stored in the `SyncFutureTask` and can eventually be picked up by the consumer of the `SyncFutureTask`.

This was causing failures in simulator because some expected exceptions like timeouts were going to the uncaught exception handler even though it was just timeouts caused by simulator induced chaos.

It's fine for these to go to the uncaught exception handler I think, but simulator shouldn't fail on expected exceptions so maybe that is what should change?

Reply:
>  simulator shouldn't fail on expected exceptions so maybe that is what should change?

That's probably the right thing to do